formulaire/src/main/java/fr/openent/formulaire/controllers/FormController.java

```diff
@@ -1068,7 +1070,53 @@ public void export(final HttpServerRequest request) {
     @ResourceFilter(ShareAndOwner.class)
     @SecuredAction(value = MANAGER_RESOURCE_RIGHT, type = ActionType.RESOURCE)
     public void shareJson(final HttpServerRequest request) {
-        super.shareJson(request, false);
+        final String id = request.params().get(ID);
+        if (id != null && !id.trim().isEmpty()) {
+            UserUtils.getUserInfos(this.eb, request, user -> {
+                if (user == null) {
+                    String message = "[Formulaire@shareJson] User not found in session.";
+                    log.error(message);
+                    unauthorized(request, message);
+                    return;
+                }
+
+                super.shareService.shareInfos(user.getUserId(), id, I18n.acceptLanguage(request), request.params().get(SEARCH), shareInfosEvt -> {
+                    if (shareInfosEvt.isLeft()) {
+                        log.error("[Formulaire@shareJson] Fail to get sharing infos : " + shareInfosEvt.left().getValue());
+                        renderInternalError(request, shareInfosEvt);
+                        return;
+                    }
+
+                    JsonObject shareInfos = shareInfosEvt.right().getValue().copy();
+                    formService.get(id, user, formEvt -> {
+                        if (formEvt.isLeft()) {
+                            log.error("[Formulaire@shareJson] Fail to get form with id " + id + " : " + formEvt.left().getValue());
+                            renderInternalError(request, formEvt);
+                            return;
+                        }
+
+                        JsonObject form = formEvt.right().getValue();
+                        if (form.getBoolean(IS_PUBLIC, false)) {
```

Comment:
maintenant que tu gères le 2eme paramètre c'est ok mais si jamais le clé contient bien "is_public" mais assignée à null, ton 2eme param le prendra pas en compte car c'est "techniquement" voulu qu'on veut null donc ca te fera avoir une NPE.

le code Boolean.TRUE te permet d'éviter ce type de soucis: https://rules.sonarsource.com/java/RSPEC-5411
De plus, comme Boolean prend en compte Null, c'est pas plus mal de le sécuriser

Reply:
Les infos sont get de la BDD et y'a une contrainte NOT NULL avec un DEFAULT FALSE sur cette column donc impossible que la valeur soit nulle